sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_endpoint_operations.py

```diff
@@ -375,15 +376,21 @@ def invoke(  # pylint: disable=too-many-locals, too-many-branches, too-many-stat
         module_logger.debug("ml_audience_scopes used: `%s`\n", ml_audience_scopes)
         key = self._credentials.get_token(*ml_audience_scopes).token
         headers[EndpointInvokeFields.AUTHORIZATION] = f"Bearer {key}"
+        headers[EndpointInvokeFields.REPEATABILITY_REQUEST_ID] = str(uuid.uuid4())
 
         if deployment_name:
             headers[EndpointInvokeFields.MODEL_DEPLOYMENT] = deployment_name
 
-        response = self._requests_pipeline.post(
-            endpoint.properties.scoring_uri,
-            json=request,
-            headers=headers,
-        )
+        while True:
```

Comment:
If we keep getting service error, it will keep trying? Will that be  like an infinite loop? Also why are we doing this?  Is there a recommendation to do this. Need to understand more.

Reply:
We should limit this to say 5 attempts - we don't want the client to hang forever in the case that the user straight up doesn't have internet connection. We should also add a backoff - assuming linear is fine here.